src/main/java/com/iota/iri/service/API.java

```diff
@@ -806,10 +804,8 @@ private AbstractResponse getBalancesStatement(final List<String> addrss, final L
 
         final Map<Hash, Long> balances = new HashMap<>();
         final int index;
-        Snapshot referenceSnapshot;
-        synchronized (instance.milestone.latestSnapshot.snapshotSyncObject) {
-            referenceSnapshot = new Snapshot(instance.milestone.latestSnapshot);
-        }
+        Snapshot referenceSnapshot = instance.milestone.latestSnapshot.get();
```

Comment:
`getBalances` doesn't work.
it fails with `java.lang.StackOverflowError` in L814.